client/src/fluid/FluidTypes.res

```diff
@@ -174,6 +174,7 @@ module AutoComplete = {
     | FPAConstructor(ID.t, string, list<ProgramTypes.Pattern.t>)
     | FPANull(ID.t)
     | FPABool(ID.t, bool)
+    | FPATuple(ID.t, ProgramTypes.Pattern.t, ProgramTypes.Pattern.t, list<ProgramTypes.Pattern.t>)
```

Comment:
There seems to be quite a lot of code here to handle the fact that we use real patterns as subcomponents of patternItems, instead of using cut down versions (like a patternItem or something like an AstRef)

Reply:
I can imagine defining more patternItems, and nesting those.

I'm not sure how AstRef is relevant here?

In any case, later commits have removed the PT references from the FPATuple definition, so there's much less "quite a lot of code here to handle the fact that we use real patterns as subcomponents of patternItems"

Reply:
though, admittedly removing the PT references from FPATuple is only temporary - that won't work once we start suggesting completions with non-blank subpatterns.

Reply:
I'll proceed with the sub-`patternItem` approach. curious how AstRefs would be useful here, though!

Reply:
AstRefs aren't useful actually, i got confused.

Reply:
removing PTs from `patternItem` _greatly_ simplified the relevant code. no conflicting IDs to worry about.